tests/test_filter_jobs.py

```diff
@@ -0,0 +1,20 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.treeherder import TreeherderPage
+
+
+@pytest.mark.nondestructive
+def test_filter_jobs(base_url, selenium):
+    """Open resultset page and filter for platform"""
+    term = u'Linux'
+    page = TreeherderPage(base_url, selenium).open()
+    page.filter_search(term)
+    assert "&filter-searchStr=Linux" in page.selenium.current_url
```

Comment:
I think a more valuable check would be that there are only Linux jobs listed. This would probably require a check that there are not just linux jobs before we filter (which is going to be dependent on data). Something like:

```python
platform = 'linux'
initial_builds = page.result_sets[0].builds
initial_platforms = [b.platform for b in initial_builds]

page.filter_by(platform)
filtered_builds = page.result_sets[0].builds
filtered_platforms = [b.platform for b in filtered_builds]
assert len(initial_builds) > len(filtered_builds)
assert all([platform.lower() in p.lower() for p in filtered_platforms])

page.clear_filter()
unfiltered_builds = page.result_sets[0].builds
unfiltered_platforms = [b.platform for b in unfiltered_builds]
assert len(initial_builds) == len(unfiltered_builds)
assert initial_platforms == unfiltered_platforms
```

This would require a `Builds` region under `ResultSet` with a `platform` property. We probably should then move `jobs` to beneath `Builds`, though we could keep a `jobs` property under `ResultSet` that returned all the jobs for all the builds.